File: src/com/ashcraftmedia/daftman/search/State.java
package com.ashcraftmedia.daftman.search;
import com.ashcraftmedia.daftman.sprite.SpriteDirection;
import com.ashcraftmedia.daftman.tile.Tile;



public class State {
	private Tile tile;
	private SpriteDirection direction;
	private int weight;
	
	public State(Tile tile, SpriteDirection direction, int weight) {
		this.tile = tile;
		this.direction = direction;
		this.weight = weight;
	}
	
	public State(Tile tile) {
		this(tile, null, 0);
	}
	
	public Tile getTile() {
		return tile;
	}
	
	public SpriteDirection getDirection() {
		return direction;
	}
	
	public int getWeight() {
		return weight;
	}
	
	public void setWeight(int weight) {
		this.weight = weight;
	}
}

File: web/js/search/State.js
export default class State {
    constructor(tile, direction = null, weight = 0) {
        this._tile = tile;
        this._direction = direction;
        this._weight = weight;
    }

    getTile() {
        return this._tile;
    }

    getDirection() {
        return this._direction;
    }

    getWeight() {
        return this._weight;
    }

    setWeight(weight) {
        this._weight = weight;
    }
}
